src/node_zlib.cc

```diff
@@ -1683,8 +1683,16 @@ static uint32_t FastCRC32(v8::Local<v8::Value> receiver,
                           uint32_t value,
                           // NOLINTNEXTLINE(runtime/references)
                           v8::FastApiCallbackOptions& options) {
-  TRACK_V8_FAST_API_CALL("zlib.crc32");
   v8::HandleScope handle_scope(options.isolate);
+  if (!data->IsArrayBufferView() && !data->IsString()) {
+    TRACK_V8_FAST_API_CALL("zlib.crc32.error");
+    THROW_ERR_INVALID_ARG_TYPE(
```

Comment:
This is already validated in the JS layer: https://github.com/nodejs/node/blob/42d0e1307b11bc2a402716142e9abc51aec73693/lib/zlib.js#L949-L955

This can just be a CHECK, as the slow callback does already.